tests/integration/test_task_pool.py

```diff
@@ -593,3 +593,368 @@ async def test_load_db_bad_platform(
             False, 1, '', 'culdee-fell-summit', '', '', '', '{}'
         ))
         assert result == 'culdee-fell-summit'
+
+
+def list_tasks(schd):
+    """Return a list of task pool tasks (incl hidden pool tasks).
+
+    Returns a list in the format:
+        [
+            (cycle, task, state)
+        ]
+
+    """
+    return sorted(
+        (itask.tokens['cycle'], itask.tokens['task'], itask.state.status)
+        for itask in schd.pool.get_all_tasks()
+    )
+
+
+@pytest.mark.parametrize(
+    'graph_1, graph_2, '
+    'expected_1, expected_2, expected_3, expected_4',
+    [
+        param(  # Restart after adding a prerequisite to task z
+            '''a => z
+               b => z''',
+            '''a => z
+               b => z
+               c => z''',
+            [
+                ('1', 'a', 'running'),
```

Comment:
Given that everything is at CP = 1, could we just test `output[1:] = expect` and not include the '1'? 

It's marginal, but I think it cuts down on the amount written. Don't care that much.